test/unit/cmap/auth/mongodb_oidc/aws_device_workflow.test.ts

```diff
@@ -0,0 +1,32 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { AwsDeviceWorkflow, Connection, MongoCredentials } from '../../../../mongodb';
+
+describe('AwsDeviceWorkFlow', function () {
+  describe('#execute', function () {
+    const workflow = new AwsDeviceWorkflow();
+
+    context('when AWS_WEB_IDENTITY_TOKEN_FILE is not in the env', function () {
+      let file;
+      const connection = sinon.createStubInstance(Connection);
+      const credentials = sinon.createStubInstance(MongoCredentials);
+
+      before(function () {
+        file = process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+        delete process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+      });
+
+      after(function () {
+        process.env.AWS_WEB_IDENTITY_TOKEN_FILE = file;
+      });
+
+      it('returns an error in the callback', async function () {
+        const result = await workflow.execute(connection, credentials).catch(error => {
+          expect(error.message).to.include('AWS_WEB_IDENTITY_TOKEN_FILE');
+        });
+        expect(result).to.equal(undefined, 'must fail without AWS_WEB_IDENTITY_TOKEN_FILE');
```

Comment:
This will always be true, because of the success case the value will be passed to the callback, not returned from the function

Reply:
Changed to use try/catch